Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -579,16 +554,51 @@ public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetBy
             .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
     }
 
-    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetPendingSubmissions(int page)
-        => await this.submissionsCommonData
-            .GetAllPending()
-            .OrderByDescending(s => s.Id)
-            .MapCollection<SubmissionForPublicSubmissionsServiceModel>()
-            .ToPagedResultAsync(DefaultSubmissionsPerPage, page);
-
     public Task<int> GetTotalCount()
         => this.submissionsData.GetTotalSubmissionsCount();
 
+    public async Task<PagedResult<SubmissionForPublicSubmissionsServiceModel>> GetSubmissions(SubmissionEnumType type, int page)
+    {
+        if (type == SubmissionEnumType.Public)
+        {
+            var user = this.userProviderService.GetCurrentUser();
+
+            if (user.IsAdminOrLecturer)
+            {
+                return await this.submissionsData.GetLatestSubmissions<SubmissionForPublicSubmissionsServiceModel>(
```

Comment:
Put `GetLatestSubmissions` on a new row. Same thing can be done for the parameters.

Reply:
it's fixed